kernel/src/hil/time.rs

```diff
@@ -1,5 +1,7 @@
 //! Hardware agnostic interfaces for counter-like resources.
 
+use crate::ReturnCode;
+
 pub trait Time {
```

Comment:
What's the reasoning for breaking out the `Time` trait? I had originally used `Counter` as the base trait because both alarms and timers must have an underlying counter. I understand the issue that if you have multiple timers based on a Counter then you may not want one of them to be able to stop the Counter. In this case, I'd suggest not making Timer (or Alarm) a subtrait of Counter, and instead have combined traits (e.g., TimerCounter).